kahuna/public/js/routes/index.js

```diff
@@ -0,0 +1,45 @@
+import angular from 'angular';
+import {searchQueryService} from '../search-query/service';
+
+// Use the name once that PR in merged in
+import '../services/api/media-api';
+
+export const indexRoute = angular.module('gr.routes.index', []);
+
+indexRoute.config(['$stateProvider', '$urlMatcherFactoryProvider',
+    function($stateProvider, $urlMatcherFactoryProvider) {
+
+        const zeroWidthSpace = /[\u200B]/g;
+        function removeUtf8SpecialChars(val) {
+            return angular.isDefined(val) && val.replace(zeroWidthSpace, '');
+        }
+        $urlMatcherFactoryProvider.type('Query', {
+            encode: val => removeUtf8SpecialChars(val),
+            decode: val => removeUtf8SpecialChars(val),
+            //call decode value that includes zero-width-space character
+            is: val => angular.isDefined(val) && !zeroWidthSpace.test(val)
+        });
+
+        $stateProvider.state('gr', {
```

Comment:
Having this one root route allows as to carry the stateParams across to other routes.
At the moment this is just between `search` and `image`.